src/main/java/com/denizenscript/depenizen/bukkit/properties/worldguard/WorldGuardNPCProperties.java

```diff
@@ -0,0 +1,101 @@
+package com.denizenscript.depenizen.bukkit.properties.worldguard;
+
+import com.denizenscript.denizen.objects.NPCTag;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.properties.Property;
+import com.denizenscript.denizencore.tags.Attribute;
+import com.denizenscript.depenizen.bukkit.objects.worldguard.WorldGuardRegionTag;
+import net.citizensnpcs.trait.waypoint.WanderWaypointProvider;
+import net.citizensnpcs.trait.waypoint.Waypoints;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+
+public class WorldGuardNPCProperties implements Property {
+
+    @Override
+    public String getPropertyString() {
+        return null;
+    }
+
+    @Override
+    public String getPropertyId() {
+        return "WorldGuardNPC";
+    }
+
+    public static boolean describes(ObjectTag object) {
+        return object instanceof NPCTag;
+    }
+
+    public static WorldGuardNPCProperties getFrom(ObjectTag object) {
+        if (!describes(object)) {
+            return null;
+        }
+        else {
+            return new WorldGuardNPCProperties((NPCTag) object);
+        }
+    }
+
+    public static final String[] handledTags = new String[] {
+            "worldguard"
+    };
+
+    public static final String[] handledMechs = new String[] {
+            "wander_worldguardregion"
+    };
+
+    public WorldGuardNPCProperties(NPCTag npc) {
+        this.npc = npc;
+    }
+
+    NPCTag npc;
+
+    @Override
+    public ObjectTag getObjectAttribute(Attribute attribute) {
+        if (attribute == null) {
+            return null;
+        }
+
+        if (attribute.startsWith("worldguard")) {
+            Waypoints waypoints = npc.getCitizen().getOrAddTrait(Waypoints.class);
+
+            // <--[tag]
+            // @attribute <NPCTag.worldguard.wander_region>
+            // @returns WorldGuardRegionTag
+            // @plugin Depenizen, WorldGuard
+            // @description
+            // Returns the worldguardRegion for the NPC's wander Waypoint Provider, if that provider is in use.
+            // -->
+            if ((attribute.startsWith("worldguard.wander_region")) && (waypoints.getCurrentProvider() instanceof WanderWaypointProvider wanderWaypointProvider)) {
```

Comment:
This is a very outdated legacy format, please do not add _new_ code in the _old_ format